File: subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>>v;
    void subset(vector<int>&nums,int i,vector<int>temp)
    {
        if(nums.size()==i)
        {
            v.push_back(temp);
            return ;
        }
        subset(nums,i+1,temp);
        temp.push_back(nums[i]);
        subset(nums,i+1,temp);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
    
        vector<int>temp;
        subset(nums,0,temp);
        return v;
    }
};

File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        ans = []
        def solve(ind,tmp):
            if ind == n:
                ans.append(tmp.copy())
                return 
            tmp.append(nums[ind])
            solve(ind+1,tmp)
            tmp.pop()
            solve(ind+1,tmp)
        solve(0,[])
        return ans